File: src/ashwini/model/params.jl
include("cell_voltage.jl")

# The model parameter dₖ₁

function dₙ₁(I, cₛₙ₀::Float64, cₛₙₗ::Float64, c₂ᵢₙ::Float64, c₂ₙ₀::Float64, c₂ₙₐ::Float64)
    F*lₙ*(I/(2*k₂ₙ(c₂ₙₐ)) - I/(2*σ₁ₙ) - 2*θ*log(c₂ᵢₙ/c₂ₙ₀)/lₙ - (Uₙ(cₛₙₗ) - Uₙ(cₛₙ₀))/lₙ)/(2*R*T) + log(jₙ₀(cₛₙₗ, c₂ᵢₙ)./jₙ₀(cₛₙ₀ , c₂ₙ₀))
end

function dₙ₁(I, cₛₙ₀::Float64, cₛₙₗ::Float64, paramsₙ)
    F*lₙ*(I/(2*k₂ₙ(paramsₙ)) - I/(2*σ₁ₙ) - 2*θ*log(c₂ᵢₙ(paramsₙ)./c₂ₙ₀(paramsₙ))/lₙ - (Uₙ(cₛₙₗ) - Uₙ(cₛₙ₀))/lₙ)/(2*R*T) + log(jₙ₀(lₙ, cₛₙₗ, paramsₙ)./jₙ₀(0, cₛₙ₀ , paramsₙ))
end

function dₚ₁(I, cₛₚ₀::Float64, cₛₚₗ::Float64, c₂ᵢₚ::Float64, c₂ₚₗ::Float64, c₂ₚₐ::Float64)
    F*lₚ*(I/(2*k₂ₚ(c₂ₚₐ)) - I/(2*σ₁ₚ) + 2*θ*log(c₂ᵢₚ/c₂ₚₗ)/lₚ - (Uₚ(cₛₚₗ) - Uₚ(cₛₚ₀))/lₚ)/(2*R*T) + log(jₚ₀(cₛₚ₀, c₂ᵢₚ)/jₚ₀(cₛₚₗ, c₂ₚₗ))
end

function dₚ₁(I, cₛₚ₀::Float64, cₛₚₗ::Float64, paramsₚ)
    F*lₚ*(I/(2*k₂ₚ(paramsₚ)) - I/(2*σ₁ₚ) + 2*θ*log(c₂ᵢₚ(paramsₚ)./c₂ₚₗ(paramsₚ))/lₚ - (Uₚ(cₛₚₗ) - Uₚ(cₛₚ₀))/lₚ)/(2*R*T) + log(jₚ₀(lₙ + lₛ, cₛₚ₀, paramsₚ)./jₚ₀(L, cₛₚₗ, paramsₚ))
end

# The model parameter dₖ₂

function dₙ₂(dₙ₁)
    dₙ₁/(log((exp(dₙ₁) - 1)/dₙ₁))
end

function dₚ₂(dₚ₁)
    dₚ₁/(log((exp(dₚ₁) - 1)/dₚ₁))
end

File: scripts/new_ashwini/model/params.m
% Include cell_voltage.m
run('cell_voltage.m')

% The model parameter d_k_1

% TODO: Fix these functions

% d_n_1 function
function result = d_n_1(I, c_s_n_0, c_s_n_l, params)
    result = F*l_n*(I/(2*k_2_n(params)) - I/(2*sigma_1_n) - 2*theta*log(c_2_in(params)./c_2_n_0(params))/l_n - (U_n(c_s_n_l) - U_n(c_s_n_0))/l_n)/(2*R*T) + log(j_n_0(l_n, c_s_n_l, params)./j_n_0(0, c_s_n_0 , params));
end

% d_p_1 function
function result = d_p_1(I, c_s_p_0, c_s_p_l, params)
    result = F*l_p*(I/(2*k_2_p(params)) - I/(2*sigma_1_p) + 2*theta*log(c_2_ip(params)./c_2_p_l(params))/l_p - (U_p(c_s_p_l) - U_p(c_s_p_0))/l_p)/(2*R*T) + log(j_p_0(l_n + l_s, c_s_p_0, params)./j_p_0(L, c_s_p_l, params));
end

% The model parameter d_k_2

% d_n_2 function
function result = d_n_2(d_n_1)
    result = d_n_1/(log((exp(d_n_1) - 1)/d_n_1));
end

% d_p_2 function
function result = d_p_2(d_p_1)
    result = d_p_1/(log((exp(d_p_1) - 1)/d_p_1));
end
